django_mongodb/query.py

```diff
@@ -105,8 +106,28 @@ def join(self, compiler, connection):
         # In the lookup stage, the reference to this column doesn't include
         # the collection name.
         rhs_fields.append(rhs.as_mql(compiler, connection).replace(f"{self.table_name}.", "", 1))
+    extra = self.join_field.get_extra_restriction(self.table_alias, self.parent_alias)
+    if extra:
+        columns = []
+        for expr in extra.leaves():
+            if hasattr(expr, "lhs") and isinstance(expr.lhs, Col):
+                columns.append((expr.lhs, len(lhs_fields)))
+                lhs_fields.append(expr.lhs.as_mql(compiler, connection))
+            if hasattr(expr, "rhs") and isinstance(expr.rhs, Col):
+                columns.append((expr.rhs, None))
```

Comment:
Why does rhs not get the same as lhs. Can it not be referenced by number still?

Reply:
You are right, it should be the same.